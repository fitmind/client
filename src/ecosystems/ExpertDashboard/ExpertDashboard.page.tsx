import React from 'react';
import { Col, Container, Row, Table } from 'react-bootstrap';
import { connect } from 'react-redux';
import { RouteComponentProps, withRouter } from 'react-router';
import styled from 'styled-components';
import { bookingInterface } from '../../interfaces/responses/expert-dashboard-response';
import { ListingInterface } from '../../interfaces/responses/listing-response';
import { fetchExpertDashboardAction, fetchExpertUserAction } from '../../redux/actions/server/server.actions';
import { ApplicationState, ConnectedReduxProps } from '../../redux/reducers/root.reducer';
import { expertDashboardInterface, expertUserInterface } from '../../redux/reducers/server-reducer/server.reducer';

interface PropsFromState {
    expertUser: expertUserInterface;
    expertDashboard: expertDashboardInterface;
}

interface PropsFromDispatch {
    fetchExpertDashboardAction: typeof fetchExpertDashboardAction;
    fetchExpertUserAction: typeof fetchExpertUserAction;
}

type ExpertDashboardPageAllProps = PropsFromDispatch & RouteComponentProps & ConnectedReduxProps & PropsFromState;

export class ExpertDashboard extends React.Component<ExpertDashboardPageAllProps> {
    public componentDidMount() {
        this.props.fetchExpertDashboardAction();
        this.props.fetchExpertUserAction();
    }

    public renderBookings = (bookings: bookingInterface[]) => {
        const rows = [];

        bookings.forEach((booking, index) => {
            rows.push(
                <tr key={booking._id}>
                    <td>{index + 1}</td>
                    <td>{booking.client}</td>
                    <td>{booking.time}</td>
                    <td>{booking.email}</td>
                    <td>{booking.listing}</td>
                    <td>{booking.price}</td>
                </tr>,
            );
        });
        return rows;
    };
    public renderListings = (listings: ListingInterface[]) => {
        const rows = [];

        listings.forEach((listing, index) => {
            rows.push(
                <tr key={listing._id}>
                    <td>{index + 1}</td>
                    <td>{listing.name}</td>
                    <td>{listing.description}</td>
                    <td>{listing.price}</td>
                </tr>,
            );
        });
        return rows;
    };

    public render() {
        const { expertUser } = this.props;
        return (
            <Container fluid>
                <Row>
                    <Col md={4} />
                    <Col>
                        <GreetingContainer>
                            <Greet>Welcome {expertUser.firstName}!</Greet>
                            <h5>
                                As an approved expert - here you will be able to create, update and remove listings.
                                View your upcoming schedule and keep records of your clients.
                            </h5>
                        </GreetingContainer>
                    </Col>
                    <Col md={4} />
                </Row>
                <Row>
                    <Col md={1} />
                    <Col md={10}>
                        <CardWrapper>
                            <h3>Current Listings Live</h3>
                            <Table responsive="md">
                                <thead>
                                    <tr>
                                        <th>#</th>
                                        <th>Listing name</th>
                                        <th>Description</th>
                                        <th>Price</th>
                                    </tr>
                                </thead>
                                <tbody>
                                    {this.props.expertDashboard &&
                                        this.props.expertDashboard.liveListings &&
                                        this.renderListings(this.props.expertDashboard.liveListings)}
                                </tbody>
                            </Table>
                            <h3>Upcoming appointments</h3>
                            <Table responsive="md">
                                <thead>
                                    <tr>
                                        <th>#</th>
                                        <th>Client</th>
                                        <th>Time</th>
                                        <th>Email</th>
                                        <th>Listing</th>
                                        <th>Price</th>
                                    </tr>
                                </thead>
                                <tbody>
                                    {this.props.expertDashboard &&
                                        this.props.expertDashboard.upcomingBookings &&
                                        this.renderBookings(this.props.expertDashboard.upcomingBookings)}
                                </tbody>
                            </Table>
                        </CardWrapper>
                    </Col>
                    <Col md={1} />
                </Row>
            </Container>
        );
    }
}

const GreetingContainer = styled.div`
    margin-top: 5rem;
    text-align: center;
`;
const Greet = styled.div`
    font-size: 4.5rem;
`;

const CardWrapper = styled.div`
    margin-top: 5rem;
`;

const mapDispatchToProps = {
    fetchExpertDashboardAction,
    fetchExpertUserAction,
};
const mapStateToProps = (state: ApplicationState) => ({
    expertUser: state.server.expertUser,
    expertDashboard: state.server.expertDashboard,
});
export default withRouter(
    connect(
        mapStateToProps,
        mapDispatchToProps,
    )(ExpertDashboard),
);
