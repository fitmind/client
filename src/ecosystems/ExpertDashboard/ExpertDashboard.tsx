import React from 'react';
import { Jumbotron } from 'react-bootstrap';

const ExpertDashboardPage: React.FC = () => (
    <Jumbotron>
        <h1>Expert Dashboard</h1>
    </Jumbotron>
);

export default ExpertDashboardPage;
