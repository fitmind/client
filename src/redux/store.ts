import { Store, createStore, applyMiddleware } from 'redux';
import { ApplicationState, storeInitialState, createRootReducer } from './reducers/root.reducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import { rootSaga } from './saga/root.saga';
import { History, createBrowserHistory } from 'history';
import { routerMiddleware } from 'connected-react-router';

// We use hash history because this example is going to be hosted statically.
// Normally you would use browser history.
export const history: History = createBrowserHistory();

export const configureStore = (initialState: ApplicationState): Store<ApplicationState> => {
    const sagaMiddleware = createSagaMiddleware();
    const composeEnhancers = composeWithDevTools({});
    let store = createStore(
        createRootReducer(history),
        initialState,
        composeEnhancers(applyMiddleware(routerMiddleware(history), sagaMiddleware)),
    );

    sagaMiddleware.run(rootSaga);
    return store;
};

const store: Store<ApplicationState> = configureStore(storeInitialState);

export default store;
