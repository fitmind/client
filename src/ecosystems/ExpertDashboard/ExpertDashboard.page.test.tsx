import React from 'react';
import { shallow } from 'enzyme';
import ExpertDashboard from './ExpertDashboard.page';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store, { history } from '../../redux/store';
import { Router } from 'react-router';

describe('Expert Dashboard page', () => {
    it('renders without crashing', () => {
        shallow(
            <Provider store={store}>
                <Router history={history}>
                    <ExpertDashboard />
                </Router>
            </Provider>,
        );
    });

    it('should match the snapshot', () => {
        const component = renderer.create(
            <Provider store={store}>
                <Router history={history}>
                    <ExpertDashboard />
                </Router>
            </Provider>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
