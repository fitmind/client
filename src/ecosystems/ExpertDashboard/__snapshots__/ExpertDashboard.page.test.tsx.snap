// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expert Dashboard page should match the snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-md-4"
    />
    <div
      className="col"
    >
      <div
        className="sc-bdVaJa hBYjom"
      >
        <div
          className="sc-bwzfXH dhFCHk"
        >
          Welcome 
          !
        </div>
        <h5>
          As an approved expert - here you will be able to create, update and remove listings. View your upcoming schedule and keep records of your clients.
        </h5>
      </div>
    </div>
    <div
      className="col-md-4"
    />
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-1"
    />
    <div
      className="col-md-10"
    >
      <div
        className="sc-htpNat fYCKZt"
      >
        <h3>
          Current Listings Live
        </h3>
        <div
          className="table-responsive-md"
        >
          <table
            className="table"
          >
            <thead>
              <tr>
                <th>
                  #
                </th>
                <th>
                  Listing name
                </th>
                <th>
                  Description
                </th>
                <th>
                  Price
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
        <h3>
          Upcoming appointments
        </h3>
        <div
          className="table-responsive-md"
        >
          <table
            className="table"
          >
            <thead>
              <tr>
                <th>
                  #
                </th>
                <th>
                  Client
                </th>
                <th>
                  Time
                </th>
                <th>
                  Email
                </th>
                <th>
                  Listing
                </th>
                <th>
                  Price
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
        </div>
      </div>
    </div>
    <div
      className="col-md-1"
    />
  </div>
</div>
`;
